### PR TITLE
Add test for one file which depends on another via goog.provide/require.

### LINKED ISSUE #30: Handle multiple inputs with dependencies
When multiple files are presented, and one of them goog.require's a symbol that is goog.provide'd by another, we should
- write one .d.ts file for each .js input
- use our global internal namespace to reference symbols from the other file.

### PR DESCRIPTION
**Pull request by alexeagle**
Write a .d.ts file for each .js input. Use the global namespace to
reference symbols between these files.

Fixes #30

### CODE REVIEW

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java [hunk 1]
```diff
@@ -463,10 +484,12 @@ public Void caseStringType() {
 
         @Override
         public Void caseObjectType(ObjectType type) {
-          if (type.isRecordType()) {
+          if (type.isNativeObjectType()) {
```
Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: dumb question, what is the diff between NativeObjectType and ObjectType?

File: src/test/java/com/google/javascript/cl2dts/require/RequireTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.runners.JUnit4;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+
+// TODO(alexeagle): should we have an explicit test class like this in each directory?
+// Or should we auto-discover them as we do for test cases in the root directory?
```
Comment (rkirov) [inline src/test/java/com/google/javascript/cl2dts/require/RequireTest.java]: auto-discover usually doesn't scale well, when we need some flexibility. Rather go for explicit over implicit.

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java (hunk 1 above)
Comment (alexeagle) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: NativeObjectType seems to be the builtin 'Object' type, as opposed to a
subtype of Object. but I don't really know

Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: I guess, something that is not a Function, interface etc. I wonder if that can simplify our isDefault check (obviously, for the next time we revisit that code).

**Comment by rkirov** [comment]
LGTM, but why does the test fail? Looks like it is not seeing one of the dependencies.

**Comment by alexeagle** [comment]
abandoning, this is old and wrong

**Pull request closed without merging**